test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterTest.scala

```diff
@@ -383,6 +383,159 @@ class CharacterTest {
+    assertEquals('ａ', Character.toLowerCase('Ａ'))
+    assertEquals('ｚ', Character.toLowerCase('Ｚ'))
+    assertEquals('［', Character.toLowerCase('［'))
+  }
+
+  @Test def toLowerCase_Int(): Unit = {
+    // Low
+    assertEquals(10, Character.toLowerCase(10)) // '\n' => '\n'
+    // Ascii
+    assertEquals(97, Character.toLowerCase(97)) // a => a
+    assertEquals(97, Character.toLowerCase(65)) // A => a
+    assertEquals(102, Character.toLowerCase(70)) // F => f
+    assertEquals(122, Character.toLowerCase(90)) // Z => z
+    // compat/ucd
+    assertEquals(0x03B2, Character.toLowerCase(0x0392)) // Β => β
+    assertEquals(0x03BC, Character.toLowerCase(0x039C)) // Μ => μ
+    // alt
+    assertEquals(257, Character.toLowerCase(257))
+    assertEquals(257, Character.toLowerCase(256))
+    assertEquals(259,Character.toLowerCase(258))
+    assertEquals(303, Character.toLowerCase(302))
+    // high
+    assertEquals(65345, Character.toLowerCase(65313))
+    assertEquals(65370, Character.toLowerCase(65338))
+    assertEquals(65339, Character.toLowerCase(65339))
+  }
+
+  @Test def toUpperCase_Char(): Unit = {
+    // Low
+    assertEquals('\n', Character.toUpperCase('\n'))
+    // Ascii
+    assertEquals('0', Character.toUpperCase('0'))
+    assertEquals('A', Character.toUpperCase('A'))
+    assertEquals('A', Character.toUpperCase('a'))
+    assertEquals('F', Character.toUpperCase('f'))
+    assertEquals('Z', Character.toUpperCase('z'))
+    // compat/ucd
+    assertEquals('Β', Character.toUpperCase('β'))
+    assertEquals('Μ', Character.toUpperCase('μ'))
+    // alt (what is this?)
+    assertEquals('Ā', Character.toUpperCase('Ā'))
+    assertEquals('Ā', Character.toUpperCase('ā'))
+    assertEquals('Ă', Character.toUpperCase('ă'))
+    assertEquals('Į', Character.toUpperCase('į'))
+    // high
+    assertEquals('Ａ', Character.toUpperCase('ａ'))
+    assertEquals('Ｚ', Character.toUpperCase('ｚ'))
+    assertEquals('［', Character.toUpperCase('［'))
+  }
+
+  @Test def toUpperCase_Int(): Unit = {
+    // Low
+    assertEquals(10, Character.toLowerCase(10)) // '\n' => '\n'
+    // Ascii
+    assertEquals(65, Character.toUpperCase(65)) // A => A
+    assertEquals(65, Character.toUpperCase(97)) // a => A
+    assertEquals(70, Character.toUpperCase(102)) // f => F
+    assertEquals(90, Character.toUpperCase(122)) // z => Z
+    // compat/ucd
+    assertEquals(0x0392, Character.toUpperCase(0x03D0)) // β => Β
+    assertEquals(0x039C, Character.toUpperCase(0x00B5)) // μ => Μ
+    // alt
+    assertEquals(256, Character.toUpperCase(256))
+    assertEquals(256, Character.toUpperCase(257))
+    assertEquals(258, Character.toUpperCase(259))
+    assertEquals(302, Character.toUpperCase(303))
+    // high
+    assertEquals(65313, Character.toUpperCase(65345))
+    assertEquals(65338, Character.toUpperCase(65370))
+    assertEquals(65339, Character.toUpperCase(65339))
+  }
+
+/*
+for(cp <- 0 until Character.MAX_CODE_POINT; if cp <= Character.MAX_VALUE) {
```

Comment:
Here we should iterate over Chars, not over code points. Chars are not code points (they are code *units*):
```suggestion
for (c <- 0.toChar until Character.MAX_VALUE) {
```